core/src/layers/prometheus.rs

```diff
@@ -0,0 +1,944 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::io;
+use std::sync::Arc;
+use std::task::Context;
+use std::task::Poll;
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::FutureExt;
+use futures::TryFutureExt;
+use log::debug;
+
+use crate::ops::*;
+use crate::raw::*;
+use crate::*;
+
+use prometheus::{
+    core::{AtomicU64, GenericCounter},
+    register_int_counter_with_registry, Histogram,
+};
+use prometheus::{exponential_buckets, histogram_opts, register_histogram_with_registry, Registry};
+
+use crate::raw::Accessor;
```

Comment:
Please group external crates with external crates, and in-crate imports with in-crate imports.

Please do list items instead of grouping them in braces.

Reply:
Oh, please avoid reviewing code format related issues :heart: